### PR TITLE
Remove sleep from tests

### PR DESCRIPTION
**Pull request by jakemcdermott**
##### SUMMARY
- Remove sleeps

### CODE REVIEW

File: awx/ui/src/components/LaunchButton/LaunchButton.test.js [hunk 1]
```diff
@@ -128,7 +125,7 @@ describe('LaunchButton', () => {
     WorkflowJobTemplatesAPI.launch.mockImplementation(async () => {
       // return asynchronously so isLaunching isn't set back to false in the
       // same tick
-      await sleep(10);
+      await new Promise((resolve) => setTimeout(resolve, 10));
```
Comment (jakemcdermott) [inline awx/ui/src/components/LaunchButton/LaunchButton.test.js:128]: Unsure if this is still needed.

Comment (AlexSCorey) [reply awx/ui/src/components/LaunchButton/LaunchButton.test.js:128]: The test passes locally without `new Promise` and `sleep`.  However, the test, which is supposed to check if the button is disabled, never asserts that `disabled === true`, so I'm not really sure what this is actually testing.

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
This all seem good to go except for that 1 line in question in the `LaunchButton` test.  If in fact it is not actually testing what it says it is we should either remove it (not my first choice) or update it.

File: awx/ui/src/components/LaunchButton/LaunchButton.test.js (hunk 1 above)
Comment (jakemcdermott) [reply awx/ui/src/components/LaunchButton/LaunchButton.test.js:128]: Good find @AlexSCorey

**Review by AlexSCorey — APPROVED** [review]
Looks good

**Comment by jakemcdermott** [comment]
@AlexSCorey - Looks like the commit to address the launch button test resulted in some new test failures. I agree that we should update that test, but since the issue wasn't introduced by this PR I'm going to leave that commit out for now and we can follow up on updating the launch button test with a separate issue: https://github.com/ansible/awx/issues/11433

**Pull request merged**